x/ibc/testing/chain.go

```diff
@@ -581,13 +576,11 @@ func (chain *TestChain) CreateTMClientHeader(chainID string, blockHeight int64,
 		Commit: commit.ToProto(),
 	}
 
-	if tmValSet != nil {
-		valSet, err = tmValSet.ToProto()
-		if err != nil {
-			panic(err)
-		}
+	valSet, err := tmValSet.ToProto()
```

Comment:
if tmValSet  == nil , there would no error raised according to this code:
```
func (vals *ValidatorSet) ToProto() (*tmproto.ValidatorSet, error) {
	if vals.IsNilOrEmpty() {
		return &tmproto.ValidatorSet{}, nil // validator set should never be nil
	}
```
I would suggest to raise an error like that return `return &tmproto.ValidatorSet{}, fmt.Errorf("Validator set is nil or empty")`

Reply:
The variable name is misleading. This is the trusted validator set which may be nil in IBC when creating a new client. I believe this will panic?

Reply:
`tmValSet == nil` is not possible, because we have `require.NotNil(chain.t, tmValSet)` in line 548